apps/web/src/utils/ui/highlightUtils.ts

```diff
@@ -153,19 +153,19 @@ export function applyHighlightsToContainer(
 
   // Filter valid highlights and sort by start offset
   const validHighlights = highlights
-    .filter((comment) => comment.highlight.isValid && comment.isValid)
-    .sort((a, b) => a.highlight.startOffset - b.highlight.startOffset);
+    .filter((comment) => comment.highlight?.isValid && comment.isValid)
+    .sort((a, b) => (a.highlight?.startOffset || 0) - (b.highlight?.startOffset || 0));
```

Comment:
this makes me nervous. I assume we could just filter out comments without startOffsets above, that would mean they're not valid. I could also see having a dedicated type, for these.